main.nf

```diff
@@ -0,0 +1,369 @@
+
+check_lesions_number_compare_t1
+  .concat(check_t1s_number)
+  .toList()
+  .subscribe{a, b -> if (a != b && b > 0)
+  error "Error ~ Some subjects have a T1w and others don't.\n" +
+        "Please be sure to have the same acquisitions for all subjects."}
+
+// Check TRKs
+check_trks
+.subscribe{a -> if (a == 0)
+    error "Error ~ No tractograms found. Please check the naming convention, your --tractograms path."
+}
+
+tractograms_for_combine.combine(atlas_for_combine)
+    .set{trk_atlases_for_decompose_connectivity}
+
+process README {
+    cpus 1
+    publishDir = params.Readme_Publish_Dir
+    tag = "README"
+
+    output:
+    file "readme.txt"
+
+    script:
+    String list_options = new String();
+    for (String item : params) {
+        list_options += item + "\n"
+    }
+
+    """
+    echo "Disconects_flow pipeline\n" >> readme.txt
+    echo "Start time: $workflow.start\n" >> readme.txt
+    echo "[Command-line]\n$workflow.commandLine\n" >> readme.txt
+    echo "[Options]\n" >> readme.txt
+    echo "$list_options" >> readme.txt
+    """
+}
+
+process Copy_Atlas {
+    cpus 1
+    publishDir = {"${params.output_dir}/"}
+    tag = "Atlas"
+
+    input:
+    set atlas_name, file(atlas_labels), file(atlas_labels_txt), file(atlas_list), file(atlas_t1) from atlas_for_copy
+
+    output:
+    file("${atlas_name}_labels.nii.gz")
+    file("${atlas_name}_labels.txt")
+    file("${atlas_name}_t1.nii.gz")
+
+    script:
+    """
+    mv ${atlas_labels} ${atlas_name}_labels.nii.gz
+    mv ${atlas_labels_txt} ${atlas_name}_labels.txt
+    mv ${atlas_t1} ${atlas_name}_t1.nii.gz
+    """
+}
+
+t1s_for_register.combine(atlas_for_registration).set{atlas_for_registration}
+
+process Register_T1 {
+    cpus params.processes_bet_register_t1
+
+    input:
+    set sid, file(t1), atlas_name, file(atlas), file(atlas_labels), file(atlas_list), file(atlas_t1) from atlas_for_registration
+
+    output:
+    set sid, atlas_name, "${sid}__output0GenericAffine.mat", "${sid}__t1_${atlas_name}_space.nii.gz" into transformation_for_registration_lesions
+    file "${sid}__t1_bet_mask.nii.gz" optional true
+    file "${sid}__t1_bet.nii.gz" optional true
+
+    script:
+    if (params.run_bet){
+    """
+        export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=$task.cpus
+        export OMP_NUM_THREADS=1
+        export OPENBLAS_NUM_THREADS=1
+        export ANTS_RANDOM_SEED=1234
+
+        antsBrainExtraction.sh -d 3 -a $t1 -e $params.template_t1/t1_template.nii.gz\
+            -o bet/ -m $params.template_t1/t1_brain_probability_map.nii.gz -u 0
+        scil_image_math.py convert bet/BrainExtractionMask.nii.gz ${sid}__t1_bet_mask.nii.gz --data_type uint8
+        scil_image_math.py multiplication $t1 ${sid}__t1_bet_mask.nii.gz ${sid}__t1_bet.nii.gz
+
+        ${params.registration_script} -d 3 -m ${sid}__t1_bet.nii.gz -f ${atlas_t1} -n ${task.cpus} -o "${sid}__output" -t ${params.registration_strategy}
```

Comment:
Shouldn't we also register tractograms to the atlas space? Or at least specify that they MUST be in the same space as the atlas for the flow to run properly?